interview-questions.md

```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
+ put/patch '/resources/:id' => 'resources#update'
+ delete '/resources/:id' => 'resources#destroy'
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: rails g generate model that will generate a model will the name, rails generate rspec it will run the rspec in the app, rails generate migration, update the data base. 
+
+  Researched answer: Rails g model will create a class that is used to represent data and interact with the database. Rails generate rspec, rspec is a ruby gem so when its generated we use it for testing. Rails migrate it updates or changes the applications database. 
+  
 
-  Researched answer:
 
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET"    location: /students          
+action: "GET"    location: /students 
+-- students#index, this will show all the students         
 
-action: "POST"   location: /students       
+action: "POST"   location: /students  
+students#create- it will create a new pice of data 
 
 action: "GET"    location: /students/new
+students#new-- it makes a new entry
 
-action: "GET"    location: /students/2  
+action: "GET"    location: /students/2 
+students#show- it will show the second data or the student with the id of 2 
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit  
+students#edit-- it will edit that piece of data
 
-action: "PATCH"  location: /students/2      
+action: "PATCH"  location: /students/2  
+update- it will update the second piece data    
 
-action: "DELETE" location: /students/2      
+action: "DELETE" location: /students/2     
+destroy-- it will destroy the second piece of data 
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+-- As a user I can create rails app (follow the commands provided to you on the Terminal to create the app) 
+
+--Use the commands to generate a model called List that will have a title for the to do list, it will have a description and priority, all will have datatypes of strings
+
+-- As a user I can visit the rails page we just made localhost:3000
+
+-- As a user I can see all the list on the rails app 
+
+
+-- As a user I can create 3 items on the to do list- two of them marked with "yes" in priority
+
+-- As a user I can see all of the items in the list with all their information 
+
+-- As a user I can retrieve the 2nd instance of the to do list and change the title 
+
+-- As a user  I can retrieve all the instances that have a "yes" in priority 
+
+-- As a user I can rename, delete and create an item on the do to list  
+
+-- As a user I can see my to do list on the rails app
+
```

Comment:
user stories are tricky after becoming a developer. Just think of how you experience web pages like Amazon, Facebook, Twitter. What do you see as a user? What can you do as a user on that page?